src/compass/utils/range_split_spectrum.py

```diff
@@ -0,0 +1,118 @@
+    high_freq_burst = burst.radar_center_frequency + half_bw
+
+    low_band_freqs = np.array([low_freq_burst, low_freq_burst + low_bw])
+    high_band_freqs = np.array([high_freq_burst - high_bw, high_freq_burst])
+    low_center_freq = low_freq_burst + low_bw / 2
+    high_center_freq = high_freq_burst - high_bw / 2
+
+    # Initialize the split-spectrum parameter object. Note, constructor
+    # requires frequency (A/B) but this is not used anywhere below
+    rdr_grid = burst.as_isce3_radargrid()
+    split_spectrum_params = splitspectrum.SplitSpectrum(
+        rg_sample_freq=burst.range_sampling_rate,
+        rg_bandwidth=burst.range_bandwidth,
+        center_frequency=burst.radar_center_frequency,
+        slant_range=rdr_grid.slant_range,
+        freq='A')
+
+    # Save the burst locally
+    burst_path = f'{scratch_path}/burst_temp.vrt'
+    burst.slc_to_vrt_file(burst_path)
+
+    # The output burst will
+    # contain 3 bands: Band #1: low-band image; Band #2 main-band image;
+    # Band #3: high-band image.
+    valid_length = burst.last_valid_line - burst.first_valid_line
+    valid_width = burst.last_valid_sample - burst.first_valid_sample
+    in_ds = gdal.Open(burst_path, gdal.GA_ReadOnly)
+    driver = gdal.GetDriverByName('ENVI')
+    out_ds = driver.Create(f'{scratch_path}/{burst.burst_id}_low_main_high',
+                           width, length, 3, gdal.GDT_CFloat32)
+
+    # Prepare necessary variables for block processing
+    lines_per_block = min(valid_length, lines_per_block)
+    num_blocks = int(np.ceil(valid_length / lines_per_block))
+
+    for block in range(num_blocks):
+        line_start = block * lines_per_block
+        if block == num_blocks - 1:
+            block_length = valid_length - line_start
+        else:
+            block_length = lines_per_block
+
+        # Read a block of valid burst data
+        burst_data = in_ds.GetRasterBand(1).ReadAsArray(
+            burst.first_valid_sample,
+            burst.first_valid_line + line_start,
+            valid_width, block_length)
+        # Get the low band sub-image and corresponding metadata
+        burst_low_data, _ = split_spectrum_params.bandpass_shift_spectrum(
+            slc_raster=burst_data, low_frequency=low_band_freqs[0],
+            high_frequency=low_band_freqs[1],
+            new_center_frequency=low_center_freq,
+            fft_size=valid_width, window_shape=burst.range_window_coefficient,
+            window_function=window_type, resampling=False
+        )
+        # Get the high sub-image and corresponding metadata
+        burst_high_data, _ = split_spectrum_params.bandpass_shift_spectrum(
+            slc_raster=burst_data, low_frequency=high_band_freqs[0],
+            high_frequency=high_band_freqs[1],
+            new_center_frequency=high_center_freq,
+            fft_size=valid_width, window_shape=burst.range_window_coefficient,
+            window_function=window_type, resampling=False
+        )
```

Comment:
@vbrancat I'm not sure if passing a subset of the SLC (the valid part) to split spectrum is needed. Can you elaborate the reason behind this implementation? Are you concerned about the high-frequencies induced by zeros at the boundaries. How concerning is that really? cutting the signal can be interpreted as shifting the signal which in turn is equivalent to introducing a phase ramp to the spectrum. My concern with the current implementation is that we are simply changing the spectrum. It can be probably avoided but would lead to unnecessarily complicating the workflow. So unless if you have a strong reason behind the current implementation, I would say we should not subset to valid region.

Reply:
@hfattahi I agree with you. Subsetting SLC could introduce incorrect bandpassing and demodulation as well. Before moving forward, we may need to check what exactly the valid region represents in the S1 data.

Reply:
@fattahi I was concerned about the high frequency components caused by the zeros in the invalid part. I guess you are suggesting to block process the SLC with its invalid values (zero) and do not worry about them?

Reply:
Correct. No need to subset.